### DISCUSSION TITLE
I Want to understand how Smart Accounts on ZKSync work

### DISCUSSION THREAD

**Discussion by deusmachinea** [zkSync-Community-Hub/zksync-developers#292] [General] [answered]
### Module Affected

- [X] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [X] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.7

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

I wanted to understand how Smart accounts work on ZKSync. From the documentation, it is clear that the smart account has these function 

- validateTransaction
- executeTransaction
- payForTransaction : if no paymaster
- prepareForPaymaster: if paymaster
- executeTransactionFromOutside

which bootloaders call to validate and execute a transaction. What I don't understand is :
- I made some transactions but couldn't see a new smart account contract deployment.
- How my address on Ethereum is same on zkSync 

### Expected Behavior

This is just an information query. I tried to look out for any forum and this is the best i could find.I

### Current Behavior

I read a lot but couldn't find the documentation for my query. 

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by dutterbutter** [comment, marked as answer]
Thanks for your inquiry! 

I definitely encourage you to check out the documentation on [Account Abstraction](https://docs.zksync.io/build/developer-reference/account-abstraction.html) for more implementation details. To answer your questions directly:

- Not exactly sure what you mean with "...made some transactions but couldn't see a new smart account contract deployment.", however, zkSync default accounts are equivalent to EoAs on Ethereum. They are however, smart accounts. You can see that implementation [here](https://github.com/matter-labs/era-contracts/blob/main/system-contracts/contracts/DefaultAccount.sol). You can also create and build your own smart account and deploy it! See a tutorial to do so [here](https://docs.zksync.io/build/tutorials/smart-contract-development/account-abstraction/daily-spend-limit.html). 

- Because the default account on zkSync is designed to operate exactly like an EoA on Ethereum, more specifically, the addresses in both networks are derived from public keys using the same cryptographic principles, specifically the Elliptic Curve Digital Signature Algorithm (ECDSA), which is standard for Ethereum. When you generate an Ethereum address, you are essentially generating a public-private key pair. The address itself is derived from the public key by taking the last 20 bytes of the keccak-256 hash of the public key. You can see the usage of ECDSA validation in the default account contract [here](https://github.com/matter-labs/era-contracts/blob/583cb674a2b942dda34e9f46edb5a9f5b696b90a/system-contracts/contracts/DefaultAccount.sol#L155).

**Reply by deusmachinea** [reply]
Thank you for your response.

The ERC4337 specifications deploy a new smart account and add an EOA as an authenticator to this smart account. During the verification phase, the smart account checks if the signature on the user operations matches that of the owner EOA.

Note that the smart account address and the EOA are different addresses.

Are you saying that EOA address stores the bytecode of the smart contract on ZKSyncEra? So, if I call a function of the smart account using the EOA address, will it succeed? How is it possible?

**Reply by dutterbutter** [reply]
ERC4337 is different from our native account abstraction, related [docs](https://docs.zksync.io/build/developer-reference/differences-with-ethereum.html#native-aa-vs-eip-4337). 

Accounts in zkSync Era can initiate transactions, like an EOA, but can also have arbitrary logic implemented in them, like a smart contract.

**Comment by crimuntilve** [comment]
In zkSync, smart accounts are implemented using a concept called "ZK Rollup". In this model, transactions are executed and validated off-chain, and only a succinct proof is submitted to the main chain (Ethereum) for verification. This allows for scalability and cost-efficiency.

To address your specific questions:

Smart Account Contract Deployment: In zkSync, smart accounts do not have their own separate on-chain contract deployment like Ethereum smart contracts. Instead, smart contracts are represented by off-chain data structures and are executed off-chain. The on-chain data is only updated when necessary, such as when creating or removing a smart contract.

Consistent Address: Your address on Ethereum and zkSync remains the same because zkSync is built as a layer 2 solution on top of Ethereum. When you deposit funds into zkSync, your Ethereum address is used to generate a corresponding zkSync address, ensuring consistency between the two chains.

If you have further questions or need more clarification, feel free to ask!

**Comment by mganlok** [comment]
Smart accounts on ZKSync are implemented as stateful smart contracts deployed on the Ethereum blockchain. When a user creates a smart account, a corresponding smart contract is deployed to Ethereum. The user's address on Ethereum remains the same on zkSync because zkSync uses a cryptographic mapping to link Ethereum addresses to zkSync addresses. This allows users to interact seamlessly with their smart accounts on zkSync using their existing Ethereum addresses.

**Comment by mancinottimoe** [comment]
Smart accounts on ZKSync are not deployed as separate contracts on the Ethereum blockchain. Instead, they are represented by accounts within the ZKSync system. The Ethereum address you use on ZKSync is the same as your Ethereum address, allowing seamless interoperability between Ethereum and ZKSync.
